drivers/storage/portworx/manifest/manifest.go

```diff
@@ -207,16 +207,17 @@ func (m *manifest) GetVersions(
 
 	cacheExpired := m.lastUpdated.Add(refreshInterval(cluster)).Before(time.Now())
 	if _, ok := provider.(*configMap); !ok && !cacheExpired && !force {
-		return m.cachedVersions.DeepCopy()
+		return m.cachedVersions.DeepCopy(), nil
 	}
 
 	// Bug: if it fails due to temporarily network issue, we should retry.
 	rel, err := provider.Get()
 	if err != nil {
-		msg := fmt.Sprintf("Using default version due to: %v", err)
-		logrus.Error(msg)
-		m.recorder.Event(cluster, v1.EventTypeWarning, util.FailedComponentReason, msg)
-		return defaultRelease(m.k8sVersion)
+		msg := fmt.Sprintf("versions from px-versions ConfigMap cannot be fetched and URL is unreachable due to: %v", err)
+		errmsg := fmt.Sprintf("StorageCluster reconciliation paused as %v", msg)
+		logrus.Error(errmsg)
+		m.recorder.Event(cluster, v1.EventTypeWarning, util.FailedComponentReason, errmsg)
```

Comment:
Now that we are returning an error and raising a warning from the caller, should we remove this?
Can you check if it raises the event twice?

Reply:
Yes this event comes up twice, one from the "SetDefaultsOnStorageCluster" where it reports defaults cannot be set on the stc due to error . another as a failed component warning. I thought we should let the failed component warning remain as it is a better representation of the issue. Should I remove it ?

Reply:
If we fail here, we won't even proceed to install the portworx pods. The failed component warning may not be relevant in that case. Let's just return the error here and let the caller raise the event.

Reply:
Okay I will remove both the logrus and recorder.event